portal/_extensions/truncatehtml.py

```diff
@@ -79,7 +79,7 @@ def next_token(self):
         try:
             char = self.input[self.counter]
             self.counter += 1
-            if char == '&':
+            if char == '&amp;':
```

Comment:
This won't work. The `char` is always a single character so this check will always fail.  I would not change this back:

```suggestion
            if char == '&':
```